lib/src/utils/block.dart

```diff
@@ -163,36 +163,35 @@ class BlockUtils {
   }
 
   static Future<bool> _setHashAndSignature(
-      AccountBlockTemplate transaction, KeyPair currentKeyPair) async {
-    transaction.hash = BlockUtils.getTransactionHash(transaction);
-    var transSig =
-        await BlockUtils._getTransactionSignature(currentKeyPair, transaction);
-    transaction.signature = transSig;
+      AccountBlockTemplate transaction, Wallet wallet) async {
+    var source = BlockUtils.getTransactionBytes(transaction);
+    transaction.hash = Hash.digest(source);
+    transaction.signature = await wallet.sign(source);
```

Comment:
`transaction.signature = await wallet.sign(transaction.hash.getBytes()!);`
Does this work if the `sign` method is left unchanged?

Reply:
This is how it worked before the change. The hashing was done outside the sign method.

Using this method, the ledger impl. can only show the hash to the user for confirmation, because it does not have access to the raw tx data.